keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/ec2/vetos/RequiredLoadBalancerVeto.kt

```diff
@@ -0,0 +1,95 @@
+package com.netflix.spinnaker.keel.ec2.vetos
+
+import com.netflix.spinnaker.keel.api.Resource
+import com.netflix.spinnaker.keel.api.ec2.OverrideableClusterDependencyContainer
+import com.netflix.spinnaker.keel.api.ec2.loadBalancersByRegion
+import com.netflix.spinnaker.keel.api.ec2.targetGroupsByRegion
+import com.netflix.spinnaker.keel.clouddriver.CloudDriverService
+import com.netflix.spinnaker.keel.clouddriver.model.AmazonLoadBalancer
+import com.netflix.spinnaker.keel.clouddriver.model.ApplicationLoadBalancerModel
+import com.netflix.spinnaker.keel.clouddriver.model.ClassicLoadBalancerModel
+import com.netflix.spinnaker.keel.core.api.DEFAULT_SERVICE_ACCOUNT
+import com.netflix.spinnaker.keel.veto.Veto
+import com.netflix.spinnaker.keel.veto.VetoResponse
+import kotlinx.coroutines.Dispatchers.IO
+import kotlinx.coroutines.withContext
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Component
+
+@Component
+class RequiredLoadBalancerVeto(
+  private val cloudDriver: CloudDriverService
+) : Veto {
+  private val log by lazy { LoggerFactory.getLogger(javaClass) }
+
+  override suspend fun check(resource: Resource<*>): VetoResponse {
+    val spec = resource.spec as? OverrideableClusterDependencyContainer<*>
+      ?: return allowedResponse()
+
+    // avoid making call to CloudDriver to get all the load balancers if we don't need to
+    if (spec.loadBalancersByRegion.isEmpty() && spec.targetGroupsByRegion.isEmpty()) {
+      return allowedResponse()
+    }
+
+    val loadBalancers = loadBalancers(spec)
+    val missingLoadBalancers = loadBalancers.checkLoadBalancers(spec)
```

Comment:
This works a bit differently to the security group veto as I'm getting all the load balancers at once. There are a couple of reasons:

1. For ALBs we actually specify the target group rather than the load balancer name, and there's no endpoint to just get target groups and no way for us to know the name of the ALB they belong to without just going through them all.
2. Security groups may not belong to the same app, load balancers have to, so we need to check security groups individually.